4.queueRange.js

```diff
@@ -35,7 +35,15 @@
 
 // Write your function here
 const range = (queue) => {
-
+    if(queue.isEmpty()) return 0;
+    let max = 0
+    let min = 0
+    while(!queue.isEmpty()){
+        max = Math.max(max, queue.peek())
+        min = Math.min(min, queue.peek())
+        queue.dequeue()
+    }
+    return max - min 
 }
```

Comment:
Beware of starting are max and min at zero. They should be -Infinity and Infinity, or the first element in the queue. Otherwise if the queue is all positives your min will falsely be 0, and vice versa for the max.